src/main/java/org/mskcc/smile/service/impl/CmoLabelGeneratorServiceImpl.java

```diff
@@ -238,7 +244,59 @@ public String generateCmoSampleLabel(SampleMetadata sampleMetadata,
 
         return getFormattedCmoSampleLabel(patientId, sampleTypeAbbreviation, paddedSampleCounter,
                 nucleicAcidAbbreviation, paddedNucAcidCounter);
-
+    }
+    
+    @Override
+    public Status generateSampleStatus(String requestId, IgoSampleManifest sampleManifest,
+            List<SampleMetadata> existingSamples) throws JsonProcessingException {
+        Status sampleStatus = new Status();
+        Map<String, String> validationReport = new HashMap<>();
+
+        if (resolveSampleTypeAbbreviation(sampleManifest) == null) {
+            validationReport.put("specimenType", "missing");
+            validationReport.put("sampleOrigin", "missing");
+            validationReport.put("sampleClass", "missing");
```

Comment:
they might not be missing in this scenario, it's possible they all exist with technically valid values but the code cannot resolve the sample type abbreviation based on the set of rules we have defined.

Reply:
based on this, maybe we could do something like 

```
validationReport.put("sample type abbreviation", "could not resolve based on specimenType, sampleOrigin, or sampleClass")
```

Reply:
I also realize that this is a new method but I think we can simply add this code to the method [generateCmoSampleLabel](https://github.com/mskcc/smile-label-generator/pull/48/files#diff-f33dcf7a186c99aa2f460dc2906e98cc2a5c5890a4d26da1049d7fd2ab1c16b0L198) and fill in the `validationReport` messages in the same places where we're [logging](https://github.com/mskcc/smile-label-generator/pull/48/files#diff-f33dcf7a186c99aa2f460dc2906e98cc2a5c5890a4d26da1049d7fd2ab1c16b0L210)